### PR TITLE
[489208] fixed refactoring of this in anon. inner classes

### PR DESCRIPTION
**Pull request by cdietrich**
[489208] fixed refactoring of this in anon. inner classes

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=489208

**Comment by cdietrich** [comment]
for test see https://github.com/eclipse/xtext-xtend/pull/178

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/serializer/SerializerScopeProvider.java [hunk 1]
```diff
@@ -323,6 +324,9 @@ protected IScope getThisOrSuperScope(XAbstractFeatureCall call, JvmType thisOrSu
 			JvmDeclaredType thisType = ((JvmMember) logicalContainer).getDeclaringType();
 			if (thisType != thisOrSuper) {
 				name = SUPER;
+				if (logicalContainer instanceof JvmConstructor && thisType instanceof JvmGenericType && ((JvmGenericType) thisType).isAnonymous()) {
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/serializer/SerializerScopeProvider.java]: I'd rather look into getNearestLogicalContainer: the arguments of the constructor call for an anon class constructor should not return the the JvmConstructor of the anon class as their logical container. They are resolved in the context of the enclosing type.

**Comment by cdietrich** [comment]
blocked by https://github.com/eclipse/xtext/issues/2260

**Pull request closed without merging**